pkg/channel/fanout/fanout_handler_test.go

```diff
@@ -208,20 +209,24 @@ func TestFanoutHandler_ServeHTTP(t *testing.T) {
 			callableServer := httptest.NewServer(&fakeHandler{
 				handler: tc.subscriber,
 			})
+			callableServerURL := &apis.URL{Scheme: "http", Host: callableServer.URL[7:]} // strip the leading 'http://'
+
 			defer callableServer.Close()
 			channelServer := httptest.NewServer(&fakeHandler{
 				handler: tc.channel,
 			})
+			channelServerURL := &apis.URL{Scheme: "http", Host: channelServer.URL[7:]} // strip the leading 'http://'
+
 			defer channelServer.Close()
 
 			// Rewrite the subs to use the servers we just started.
 			subs := make([]eventingduck.SubscriberSpec, 0)
 			for _, sub := range tc.subs {
 				if sub.SubscriberURI == replaceSubscriber {
-					sub.SubscriberURI = callableServer.URL[7:] // strip the leading 'http://'
+					sub.SubscriberURI = callableServerURL
```

Comment:
This works; I was actually suggesting that you could write (on line 212):

```golang
replaceSubscriber.Host = callableServer.URL[7:]  // strip the leading 'http://'
replaceChannel.Host = channelServer.URL[7:]  // strip the leading 'http://'
```

And then avoid this loop altogether and just use `tc.subs` below.

Reply:
My suggestion for the `_types.go` file makes this suggestion obsolete; if we're storing a concrete type (rather than a shared pointer), we'll need to update each struct directly. Given how changing the types from pointer to concrete avoids several other classes of bugs, I'd be inclined to switch to concrete `apis.URL` types and keep this loop.

The data race is probably because you're replacing this pointer multiple times: once for each test. If the tests are run in parallel in the same process, those replacements will be racing if the tests are run in parallel.